DESIGN.md

```diff
@@ -0,0 +1,45 @@
+# Design
+
+The goal is to provide a lightweight, modular, and extensible library
+that is approachable and powerful. We will also provide useful
+documentation and examples which are type-checked by the compiler to
+ensure correctness.
+
+Cats will be designed to use modern *best practices*:
+
+ * [simulacrum](https://github.com/mpilquist/simulacrum) for minimizing type class boilerplate
+ * [machinist](https://github.com/typelevel/machinist) for optimizing implicit operators
+ * [scalacheck](http://scalacheck.org) for property-based testing
+ * [discipline](https://github.com/typelevel/discipline) for encoding and testing laws
+ * [kind-projector](https://github.com/non/kind-projector) for type lambda syntax
+ * [algebra](https://github.com/non/algebra) for shared algebraic structures
+ * ...and of course a pure functional subset of the Scala language.
+
+(We also plan to support [Miniboxing](http://scala-miniboxing.org) in a branch.)
+
+Currently Cats is experimenting with providing laziness via a type
+constructor (`Eval[_]`), rather than via ad-hoc by-name
+parameters. This design may change if it ends up being impractical.
+
+The goal is to make Cats as efficient as possible for both strict and
+lazy evaluation. There are also issues around by-name parameters that
+mean they are not well-suited to all situations where laziness is
+desirable.
+
+### Modules
+
+Cats will be split into modules, both to keep the size of the
+artifacts down and also to avoid unnecessarily tight coupling between
+type classes and data types.
+
+Initially Cats will support the following modules:
+
+ * `macros`: Macro definitions needed for `core` and other projects.
+ * `core`: Definitions for widely-used type classes and data types.
```

Comment:
@non are you planning on adding this? I'm happy to add it in a followup PR if that's easier for you.